agent/command/git.go

```diff
@@ -733,14 +733,26 @@ func (c *gitFetchProject) fetchModuleSource(ctx context.Context,
 		dir:      moduleBase,
 		method:   cloneMethod,
 	}
-	// Module's location takes precedence over the project-level clone
-	// method.
+	// If the module repo is using the deprecated ssh cloning method, extract the owner
+	// and repo from the string and construct a https cloning link manually.
+	// This is a temporary workaround which will be removed once users have switched over.
 	if strings.Contains(opts.location, "git@github.com:") {
-		opts.method = evergreen.CloneMethodLegacySSH
-	} else if opts.method == evergreen.CloneMethodOAuth {
+		owner, repo, err := thirdparty.ParseModuleLocation(opts.location)
```

Comment:
Can we add a log here explaining that we're manually converting the SSH cloning to OAuth? I feel like this inevitably may break some clones on older distros, in which case we'd need some sort of way to notify why a task's behavior has suddenly changed.

Reply:
We only do ssh cloning on modules right now. So for the distro that is broken (amazon1-2018-small), we won't be able to get the project either, so the whole patch will fail anyway. I therefore don't expect this to break anything that's not already broken. 

I think it's a good idea to log anyway though.  

Re 

> Should we add this as a temporary validation warning as well?
I have a separate ticket to figure out the best way to warn users.